src/Hedgehog/Extras/Test/File.hs

```diff
@@ -273,14 +277,25 @@ assertIsYamlFile fp = GHC.withFrozenCallStack $ do
     Right _ -> return ()
     Left msg -> H.failMessage GHC.callStack msg
 
--- | Checks if all files gives exists. If this fails, all files are deleted.
+-- | Asserts that the given file exists.
+assertFileExists :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m ()
+assertFileExists file = do
+  exists <- H.evalIO $ IO.doesFileExist file
+  unless exists $ H.failWithCustom GHC.callStack Nothing (file <> " has not been successfully created.")
+
+-- | Asserts that all of the given files exist.
 assertFilesExist :: (MonadTest m, MonadIO m, HasCallStack) => [FilePath] -> m ()
-assertFilesExist [] = return ()
-assertFilesExist (file:rest) = do
+assertFilesExist files = for_ files assertFileExists
+
+-- | Asserts that the given file is missing.
+assertFileMissing :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m ()
+assertFileMissing file = do
   exists <- H.evalIO $ IO.doesFileExist file
-  if exists
-    then GHC.withFrozenCallStack $ assertFilesExist rest
-    else H.failWithCustom GHC.callStack Nothing (file <> " has not been successfully created.")
+  when exists $ H.failWithCustom GHC.callStack Nothing (file <> " has not been successfully created.")
```

Comment:
I think the error message should be something like:
```
" has not been successfully removed."
````

Reply:
Reworded.  Thanks!